ci/ci

```diff
@@ -213,230 +400,475 @@ def check_for_existing_image():
+        val = val.replace('=', '-')
+        val = val.replace(':', '-')
+        potential_list.append(val)
+    current_file_services_list = [i[:-2] for i in current_file_services_list]
+    # compare without platform names
+    current_file_services_list = [
+        i.split('-', 1)[0] for i in current_file_services_list]
+    potential_list = [i.split('-', 1)[0] for i in potential_list]
+    potential_list.sort()
+    current_file_services_list.sort()
+    if current_file_services_list != potential_list:
+        build_compose_file(node_config)
+        return True
+    else:
+        return False
+
+
+def setup_config_dir():
+    '''
+    Initializes config directory and files for ci
+    '''
+    command_path = os.path.join(ci_dirname, MACROS['CONFIG_DIR'])
+    if not os.path.exists(command_path):
+        os.mkdir(command_path)
+    target_path = os.path.join(command_path, MACROS['CONF_JSON_FILE'])
+    if not os.path.exists(target_path):
+        value = '{ "configure": "--prefix=/opt/pbs \
+            --enable-ptl", "tests" : "-t SmokeTest" }'
+        write_to_file(target_path, value)
+    target_path = os.path.join(command_path, MACROS['CONFIGURE_OPT_FILE'])
+    if not os.path.exists(target_path):
+        value = "--prefix=/opt/pbs --enable-ptl"
+        write_to_file(target_path, value)
+    target_path = os.path.join(command_path, MACROS['BENCHPRESS_OPT_FILE'])
+    if not os.path.exists(target_path):
+        value = "--tags=smoke"
+        write_to_file(target_path, value)
+    target_path = os.path.join(ci_dirname, 'docker-compose.json')
+    if not os.path.exists(target_path):
+        build_compose_file('server=centos:7')
+        run_cmd('docker-compose -f docker-compose.json down --remove-orphans')
+
+
+def delete_ci():
+    '''
+    Takes backup of logs and deletes running containers.
+    '''
+    services = get_services_list()
+    if len(services) != 0:
+        build_compose_file(nodes=get_current_setup())
+        cmd = '/src/etc/killit.sh backup'
+        run_docker_cmd(cmd, only_run_on='server')
+        log_warning('Removed logs file')
+        log_info('backup files can be found in ' + build_log_path)
     else:
-        pbs_dirname = find_ci()
-        pbs_dirname = os.path.join(pbs_dirname, 'ci')
+        log_info('No running container found, nothing to backup')
     try:
-        os.chdir(pbs_dirname)
+        os.chdir(ci_dirname)
+        run_cmd(
+            "docker-compose -f docker-compose.json down --remove-orphans")
+        log_info(
+            "done delete container and services")
     except Exception as e:
-        log_error(str(e))
-        log_error("Was correct pbs path provided?")
+        log_error("Failed to destroy container and services: " + e)
+
+
+def parse_params(params_list):
+    '''
+    Update given params and run ci
+    '''
+    if params_list[0] != 'called':
+        container_running = False
+        conf_opts = load_conf()
+        for set_opts in params_list:
```

Comment:
if the user provides both os and nodes - we will call docker_compose twice

Reply:
There is no scenario i can see where os and nodes both are required; in any case it will overwrite the params with the latter called command, not breaking anything.